app/views/unattended/provisioning_templates/snippet/first_boot_setup.erb

```diff
@@ -0,0 +1,55 @@
+<%#
+kind: snippet
+name: first_boot_setup
+model: ProvisioningTemplate
+snippet: true
+description: |
+  Post replacement for both systemd and non-systemd platforms
+-%>
+<%
+  os_major = @host.operatingsystem.major.to_i
+  rhel_compatible = @host.operatingsystem.family == 'Redhat' && @host.operatingsystem.name != 'Fedora'
+  has_systemd = (@host.operatingsystem.name == 'Fedora' && os_major >= 20) || (rhel_compatible && os_major >= 7) || (@host.operatingsystem.name == 'Ubuntu' && os_major >= 15) || (@host.operatingsystem.name == 'Debian' && os_major >= 8)
+  initd_script = """
+  #! /bin/bash
+
+  ### BEGIN INIT INFO
+  # Provides:          foo
+  # Required-Start:    $local_fs $network
+  # Required-Stop:     $local_fs
+  # Default-Start:     2 3 4 5
+  # Default-Stop:      0 1 6
+  # Short-Description: foo service
+  # Description:       Run Foo service
+  ### END INIT INFO
+
+  # Source function library
+  . /etc/rc.d/init.d/functions
+
+  ##Service start/stop functions##
+  start() {
+      /tmp/first_boot_script
+      chkconfig --del first_boot_setup
+  }
+
+  start
+
+  exit 5
+  esac
+  exit $?
+"""
+%>
+<% if has_systemd -%>
+  <%= save_to_file('/etc/systemd/system/first_boot_service.service', snippet('first_boot_service')) %>
+  <%= save_to_file('/tmp/first_boot_script', snippet('built', :variables => { :endpoint => 'built', :method => 'POST', :body_file => '/root/install.post.log' })) %>
+  systemctl enable first_boot_service
+  chmod +x /tmp/first_boot_script
+<% else %>
+  <%= save_to_file('/tmp/first_boot_script', snippet('built', :variables => { :endpoint => 'built', :method => 'POST', :body_file => '/root/install.post.log' }) +
+    "\nmv /tmp/first_boot_script /tmp/first_boot_script.disabled") %>
```

Comment:
Why is this here? Why not on the new line as a standalone command?

Reply:
I am not sure what do you mean. putting it in a new line would override the file. 
I need this snippet: `snippet('built', :variables => { :endpoint => 'built', :method => 'POST', :body_file => '/root/install.post.log'}) `. 
And I need this line: `mv /tmp/first_boot_script /tmp/first_boot_script.disabled` at the end of the same file.